src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -487,23 +490,69 @@ class RequiredMandatoryAttribute:
                         success = False
                         continue
 
+        self.print_step(3, "Validate the global attributes are in range and do not contain duplicates")
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
```

Comment:
I know this does not really matter in the grand scheme of things, but we are iterating over this list a lot of times in this script 😅 I guess this is the result of having these separated steps in the test plans, but we could properly do most of the checks in one iteration if we really wanted, but it is most likely not worth the effort to change this now 🤷🏻‍♂️

Reply:
yup - this is 100% a relic of the test harness and test plan setup. Believe me, I also think it's daft.